### DISCUSSION TITLE
Ray information

### DISCUSSION THREAD

**Discussion by IBRAAHHIIMM** [NVlabs/sionna#551] [General]
Hi, 
I wanted to get the ray information which is antenna agnostic. I might be wrong about the documentation and it might be related to my lack of knowledge in antennas. scene.compute_paths() generates the path information and we can access a,tau via paths.a, paths.tau. However, the path information is specific to the antenna configuration that we define. I need the information on rays for point sources at TX and RX. I can generate such information when I set TX and RX antennas to "iso" with a single polarization. With dual-polarized antennas, I still get num_rx_ant, num_tx_ant number of "a" but single "tau" information, which is the same for specific elements. The information that I get through "V" polarization and "H" polarization is different for "a" values. The order of "a" values with respect to path indices is different for "V" and "H". It would be great if you could help me with the correct setup for getting antenna agnostic ray information in Sionna.

**Comment by SebastianCa** [comment]
Hi @IBRAAHHIIMM,
I assume you’ve set `synthetic_array=True`. This means that the ray tracer only traces once from the center of the antenna array and applies synthetic phase shifts, resulting in a single value for `tau`. For more details, please refer to issues #547 and #491.

Please note that polarization also occurs during wave propagation (e.g., in [scattering](https://nvlabs.github.io/sionna/em_primer.html#equation-xpd)), so it’s not just an antenna property. If you’re considering using a cross-polarized receive antenna, combining both inputs could be an option, depending on your specific goals.

**Reply by IBRAAHHIIMM** [reply]
Hi @SebastianCa ,
Thanks for the prompt reply! Yes, I set synthetic_array=True. I understood all the logic behind how it works. The point of my question is the consideration of the multipath characteristics of the environment for a particular frequency. The multipath characteristics or power angular spectrum of a specific point in a static scene should be the same regardless of the antenna polarization if I am not mistaken. I will try to read more about the effect of polarization. Here is my observation from a simple TX and RX with "iso" pattern for different polarizations.
![image](https://github.com/user-attachments/assets/3b1a0342-0784-422e-a67a-d669250f856a)
The AoD's and AoA's are the same for all paths with synthetic_array = True. My guess is there should be slight deviations if it is not set to True for far-field approximation. 
![image](https://github.com/user-attachments/assets/10e9a004-c4f1-4546-85a6-23b63f7ff926)
Let me know if you have any comments on this and if the plots for "a" make sense with the intuition of polarization. Thanks!

**Reply by faycalaa** [reply]
Hi @IBRAAHHIIMM ,

What you observe in the first figure is due to reflection altering the field's polarization. This explains why paths interacting with the scene have different attenuation coefficients, even when both antennas use the same polarization type.

**Reply by IBRAAHHIIMM** [reply]
Hi @faycalaa,
I had the same interpretation after a little research about the effects of reflection, scattering, and diffraction on polarization. Thanks for the response! Is it possible to integrate a custom antenna pattern of which we know the E field and H field components in the global coordinate system into Sionna to generate "a", and "tau"?

**Reply by faycalaa** [reply]
To gain a better understanding of how antennas are modeled in Sionna, you can check out the [EM Primer](https://nvlabs.github.io/sionna/em_primer.html).

Sionna supports [the implementation of custom antenna patterns](https://github.com/NVlabs/sionna/discussions/409). Note that the delays `tau` are independent of the antenna pattern, as they only depend on the path length.

Also, could using a dual-polarized antenna (`VH` or `cross`) at the receiver be what you're looking for? This would ensure that the two components of the incoming electric field are captured.
